### PR TITLE
Update keystone client creation to use sessions

### PR DESCRIPTION
**Pull request by lihkin213**
Ansible modules for federation are failing on master deploy because keystoneauth1 moved from 2.9.0 to 2.11.1

### CODE REVIEW

File: library/keystone_role.py [hunk 1]
```diff
@@ -19,9 +19,11 @@
 
 
 try:
+    from keystoneauth1 import session
```
Comment (j2sol) [inline library/keystone_role.py]: Is there a reason we're still using this module, instead of the `os_keystone_role` module we use elsewhere?

**Comment by jamielennox** [comment]
>  2.9.0 to 2.11.1

Why did this matter? There shouldn't have been any backwards incompatible changes in this release.

**Comment by j2sol** [comment]
Our modules were coded against a very old standard, and started failing when we changed what the catalog listed for the API endpoint (it started using an unversioned endpoint URL). The module did not construct an auth object in a way that was compatible with the endpoint URL changing, and it would attempt to discover the endpoint from the catalog. By switching to using a session, the URL used to get a token was used later to do keystone actions.

File: library/keystone_role.py (hunk 1 above)
Comment (lihkin213) [inline library/keystone_role.py]: There is no reason to use this.  I will push another PR to address this.

**Comment by lihkin213** [comment]
@j2sol can this be merged or you have additional questions. I will address your question about keystone_role with another PR once this is merged.

**Comment by j2sol** [comment]
Oh, I thought you were going to push a change in this PR to not need the obsolete module. I'll merge this.

**Pull request merged**